x509/cert.go

```diff
@@ -30,3 +33,35 @@ func ParseCertificatePEM(data []byte) ([]*x509.Certificate, error) {
 	}
 	return certs, nil
 }
+
+// ValidateCertChain takes an ordered certificate chain and validates issuance from leaf to root
+func ValidateCertChain(certChain []*x509.Certificate) error {
```

Comment:
This method only check if the cert chain is correctly formed, not that it chains to an trusted root certificate in trust store correct? Where will we have that logic. Also is this the method where we will add additional cert chain constraint check (KU, EKU, path len)?

We should create an issue to track that work and add TODO with the issue link, if that logic will be included here.